### ISSUE TITLE
VSCode implicitly import java.lang.StringTemplate.STR which conflict with org.yaml.snakeyaml.nodes.Tag.STR

### ISSUE THREAD

**Issue by renanfranca** [redhat-developer/vscode-java#3510]
VSCode implicitly import java.lang.StringTemplate.STR which conflict with org.yaml.snakeyaml.nodes.Tag.STR

##### Environment
- Operating System: Windows 10 
- JDK version: 21
- Visual Studio Code version: 1.86.2
- Java extension version: v1.28.1

##### Steps To Reproduce
1. Clone the open source project https://github.com/jhipster/jhipster-lite
2. Import the java project
3. Go to file tech.jhipster.lite.module.infrastructure.secondary.YamlFileSpringPropertiesHandler.java
4. See the compilation error (`Type mismatch: cannot convert from StringTemplate.Processor<String,RuntimeException> to Tag`)

[Screen shot]
![image](https://github.com/redhat-developer/vscode-java/assets/6011616/05585b88-26b1-4262-8aa5-256eef406099)

##### Current Result
I have to change the code only to run the project:
FROM:
```java 
  private Node buildScalarNode(Object value) {
    Tag tag = STR;
    if (value instanceof Integer || value instanceof Long) {
      tag = INT;
    } else if (value instanceof Double || value instanceof Float) {
      tag = FLOAT;
    } else if (value instanceof Boolean) {
      tag = BOOL;
    }

    return new ScalarNode(tag, value.toString(), null, null, DumperOptions.ScalarStyle.PLAIN);
  }
```

TO:

```java 
  private Node buildScalarNode(Object value) {
    Tag tag = Tag.STR;
    if (value instanceof Integer || value instanceof Long) {
      tag = INT;
    } else if (value instanceof Double || value instanceof Float) {
      tag = FLOAT;
    } else if (value instanceof Boolean) {
      tag = BOOL;
    }

    return new ScalarNode(tag, value.toString(), null, null, DumperOptions.ScalarStyle.PLAIN);
  }
  ```
##### Expected Result
Is there a configuration to solve this problem without need to explicitly use Tag.STR?

**Comment by rgrunber** [comment]
Reproducible in Eclipse & `javac` !

**Eclipse**
![image](https://github.com/redhat-developer/vscode-java/assets/1417342/0cec86a8-04ee-4810-87fb-d0188029fb1a)


**Tag.java**
```java
package org.example;
public class Tag {
	public static final String STR = "test";
}
```

**Main.java**
```java
package org.sample;
import static org.example.Tag.STR;
public class Main {
	public static void main(String[] args) {
		String value = STR;
		System.out.println(value);
	}
}
```

```bash
/usr/lib/jvm/java-21-openjdk/bin/javac --enable-preview --release 21 org/sample/Main.java 
org/sample/Main.java:5: error: reference to STR is ambiguous
		String value = STR;
		               ^
  both variable STR in StringTemplate and variable STR in Tag match
1 error

```

I think this comes down to how String Templates are defined in ([JEP 430](https://openjdk.org/jeps/430)). It says :

> STR is a public static final field that is automatically imported into every Java source file.

So `STR` becomes a kind of "restricted keyword". I think it always needs to be fully qualified now.

**Comment by fbricon** [comment]
@rgrunber javac doesn't fail without the "--enable-preview" flag. Neither does IJ IDEA. So the bug here is to expose STR when preview features are disabled. Definitely an upstream JDT issue.

**Comment by fbricon** [comment]
> But what happens when the feature becomes finalized (ie. Java 23?). Then we can't protect them from the name clash.

Yes, but that's a Java 23 problem.

**Comment by rgrunber** [comment]
You're right. I forgot that non-managed projects (like Maven, Gradle) don't have preview features enabled by default. If preview features are disabled, and this is occuring then it's a bug in JDT Core.

**Comment by fbricon** [comment]
JEP 430 is dead. STR is no longer implicitly injected by ECJ into the compilation. We're good for now.
